internal/events/events.go

```diff
@@ -0,0 +1,79 @@
+package events
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/crashtracker"
+)
+
+var (
+	ErrTopicRequired    = errors.New("message topic is required")
+	ErrKeyRequired      = errors.New("message key is required")
+	ErrTenantIDRequired = errors.New("message tenant ID is required")
+	ErrTypeRequired     = errors.New("message type is required")
+	ErrDataRequired     = errors.New("message data is required")
+)
+
+type Message struct {
+	Topic    string `json:"topic"`
+	Key      string `json:"key"`
+	TenantID string `json:"tenant_id"`
+	Type     string `json:"type"`
+	Data     any    `json:"data"`
+}
+
+func (m Message) String() string {
+	return fmt.Sprintf("Topic: %s - Key: %s - Type: %s - Tenant ID: %s", m.Topic, m.Key, m.Type, m.TenantID)
+}
+
+func (m Message) Validate() error {
+	if m.Topic == "" {
+		return ErrTopicRequired
+	}
+
+	if m.Key == "" {
+		return ErrKeyRequired
+	}
+
+	if m.TenantID == "" {
+		return ErrTenantIDRequired
+	}
+
+	if m.Type == "" {
+		return ErrTypeRequired
+	}
+
+	if m.Data == nil {
+		return ErrDataRequired
+	}
+
+	return nil
+}
+
+type Producer interface {
+	WriteMessages(ctx context.Context, messages ...Message) error
+	Close() error
+}
+
+type Consumer interface {
+	RegisterEventHandler(ctx context.Context, eventHandlers ...EventHandler) error
+	ReadMessage(ctx context.Context) error
+	Close() error
+}
```

Comment:
nit: A couple of things with this interface: 
1. `ReadMessage()` should probably be `ReadAndHandleMessage()`. Because we're also invoking the handler. 
2. `ReadMessage()` doesn't need to be exposed, it should be an implementation detail. 
3. Consume(..) should probably be on this interface. 

So my final interface would look like : 
```go
type Consumer interface {
	RegisterEventHandler(ctx context.Context, eventHandlers ...EventHandler) error
	Consume(ctx context.Context, consumer Consumer, crashTracker crashtracker.CrashTrackerClient)
	Close() error
}
```
This is how it would be used : 
```go
pingPongConsumer := events.NewKafkaConsumer(brokers, "ping-pong", consumerGroupID)
err := pingPongConsumer.RegisterEventHandler(ctx, events.PingPongEventHandler{})
defer pingPongConsumer.Close()
go pingPongConsumer.Consume(ctx, crashTrackerClient)
```

Based on this contract `Read()` or `ReadAndHandle()` doesn't need to be exposed.

Reply:
The `Consume` method should not be something from the `Consumer` interface, IMO. The main problem is that we'll reimplement the same logic for every consumer on this method. But, I agree that we should add the `crashTracker` on the Consumer. But it can be something as an attribute, not a function argument.

Reply:
changed: [9ee7d22](https://github.com/stellar/stellar-disbursement-platform-backend/pull/119/commits/9ee7d22a1d0d32495233dfe75be43cc790dba533)